com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs

```diff
@@ -51,24 +52,303 @@ public static VisibleLightAxisAndPosition GetAxisAndPosition(this VisibleLight v
-        Count
+        // 'xrViewCount' and 'maxEntityCountPerCategory' control the size of the memory allocation.
+        public BoundedEntityCollection(int xrViewCount, int[] maxEntityCountPerCategory)
+        {
+            Debug.Assert(xrViewCount > 0);
+            Debug.Assert(maxEntityCountPerCategory.Length == (int)BoundedEntityCategory.Count);
+
+            m_MaxEntityCount = 0;
+
+            for (int i = 0, n = (int)BoundedEntityCategory.Count; i < n; i++)
+            {
+                m_EntityCountPerCategory[i] = 0;
+
+                m_MaxEntityCount += maxEntityCountPerCategory[i];
+            }
+
+            m_TotalEntityCount = 0;
+
+            m_Views = new ViewDependentData[xrViewCount];
+
+            for (int i = 0, n = xrViewCount; i < n; i++)
+            {
+                m_Views[i] = new ViewDependentData();
+
+                m_Views[i].punctualLightData   = new List<LightData>();
+                m_Views[i].areaLightData       = new List<LightData>();
+                m_Views[i].reflectionProbeData = new List<EnvLightData>();
+                m_Views[i].decalData           = new List<DecalData>();
+                m_Views[i].densityVolumeData   = new List<DensityVolumeData>();
+                m_Views[i].entityBounds        = new List<FiniteLightBound>();
+                m_Views[i].entitySortKeys      = new ulong[m_MaxEntityCount];
+            }
+
+            for (int i = 0, n = (int)BoundedEntityCategory.Count; i < n; i++)
+            {
+                m_EntityDataBufferPerCategory[i] = new ComputeBuffer(maxEntityCountPerCategory[i] * xrViewCount, s_EntityDataSizesPerCategory[i]);
+            }
+
+            m_EntityBoundsBuffer = new ComputeBuffer(m_MaxEntityCount * xrViewCount, Marshal.SizeOf(typeof(FiniteLightBound)));
+        }
+
+        // Reset the counters.
+        public void Reset()
+        {
+            for (int i = 0, n = (int)BoundedEntityCategory.Count; i < n; i++)
+            {
+                m_EntityCountPerCategory[i] = 0;
+            }
+
+            m_TotalEntityCount = 0;
+
+            for (int i = 0, n = m_Views.Length; i < n; i++)
+            {
+                m_Views[i].punctualLightData.Clear();
+                m_Views[i].areaLightData.Clear();
+                m_Views[i].reflectionProbeData.Clear();
+                m_Views[i].decalData.Clear();
+                m_Views[i].densityVolumeData.Clear();
+                m_Views[i].entityBounds.Clear();
+
+                // Since we reset 'm_TotalEntityCount', we can leave 'entitySortKeys' alone.
+            }
+
+            // No need to clear ComputeBuffers, we will just overwrite the contents.
+        }
+
+        // Release the resources that are NOT automatically managed by C#.
+        // TODO: call this somewhere.
```

Comment:
Frightening todo here :)

Reply:
Maybe call it from CleanupLightLoop? Not sure.